pkg/controller/seed-controller-manager/kubernetes/pending.go

```diff
@@ -113,3 +118,19 @@ func (r *Reconciler) ensureClusterNetworkDefaults(ctx context.Context, cluster *
 		}
 	})
 }
+
+// ensureEtcdLauncherFeatureFlag will apply seed controller etcdLauncher setting on the cluster level
+func (r *Reconciler) ensureEtcdLauncherFeatureFlag(ctx context.Context, cluster *kubermaticv1.Cluster) error {
+	return r.updateCluster(ctx, cluster, func(c *kubermaticv1.Cluster) {
+		if r.features.EtcdLauncher { // enabled at the controller level
+			// we only modify the cluster feature flag if it's not explicitly set, regardless if the value
+			_, set := c.Spec.Features[kubermaticv1.ClusterFeatureEtcdLauncher]
+			if !set {
```

Comment:
personally, I find it more elegant if you get the `set` value and do the check it in one line:
```
if _, set := c.Spec.Features[kubermaticv1.ClusterFeatureEtcdLauncher]; set {
...
```